### PR TITLE
Optimize status code check

### PR DESCRIPTION
**Pull request by dmitrydprog**
image

image

### CODE REVIEW

File: storage/remote/client.go [hunk 1]
```diff
@@ -236,7 +236,7 @@ func (c *Client) Store(ctx context.Context, req []byte, attempt int) error {
 		}
 		err = fmt.Errorf("server returned HTTP status %s: %s", httpResp.Status, line)
 	}
-	if httpResp.StatusCode/100 == 5 ||
+	if httpResp.StatusCode >= 500 ||
```
Comment (machine424) [inline storage/remote/client.go:239]: What if the server is using some non standard HTTP status codes > 599?

Comment (dmitrydprog) [reply storage/remote/client.go:239]: Anyway, we expect only 2xx status, other status doesn't matter. 5xx, 6xx, 7xx, general not is 2xx

Comment (dmitrydprog) [reply storage/remote/client.go:239]: In ideal solution, check on 529 status and then try read Retry-After header, other just skip and set defaultBackoff

Comment (machine424) [reply storage/remote/client.go:239]: For me this changes the behavior as the client is only supposed to retry is the receiver sends a 5xx https://prometheus.io/docs/concepts/remote_write_spec/#retries-backoff

**Pull request closed without merging**